### PR TITLE
Rewrite for GHES to GHES sync and Dockerize

### PR DESCRIPTION
**Pull request by stoe**
Allow syncing from one (source) GitHub Enterprise Server to another (target) GitHub Enterprise Server organization.

### CODE REVIEW

**Review by pmanlukas — APPROVED** [review]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
+FROM ruby:2.6.6-slim
+
+ARG ssh_prv_key
+ARG ssh_pub_key
+ARG source_ghes_domain
+ARG target_ghes_domain
+
+WORKDIR /app
+ADD . /app/
```
Comment (peter-murray) [inline Dockerfile:9]: This should be moved down after the more "static" `RUN` commands that follow so as when iterating locally, we are not rebuilding all the following layers due to a code change

Comment (pmanlukas) [inline Dockerfile:9]:
```suggestion
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    chmod 0700 /root/.ssh && \
+    ssh-keyscan ghe.io > /root/.ssh/known_hosts \
+    ssh-keyscan github.stoelzle.me > /root/.ssh/known_hosts
+
+# Add the keys and set permissions
+RUN echo "$ssh_prv_key" > /root/.ssh/id_rsa && \
+    echo "$ssh_pub_key" > /root/.ssh/id_rsa.pub && \
+    chmod 600 /root/.ssh/id_rsa && \
+    chmod 600 /root/.ssh/id_rsa.pub
+
```
Comment (pmanlukas) [inline Dockerfile:29]:
```suggestion

WORKDIR /app
ADD Gemfile Gemfile.lock hubsync.rb /app/

```

Comment (pmanlukas) [reply Dockerfile:29]: I would move it here as @peter-murray suggested and also copy only the necessary files instead of all of them

**Review by pmanlukas — CHANGES_REQUESTED** [review]

**Pull request merged**